### DISCUSSION TITLE
Using same instance of CefSharp browser across all view models in MVVM architecture

### DISCUSSION THREAD

**Discussion by Dineshsing** [cefsharp/CefSharp#4688] [Q&A] [answered]
Hi,
I'm relatively new to MVVM, CefSharp and JS as well. 
Problem statement: I've a C# WPF (MVVM) application where I'm using Echarts to plot some stuff in a Cef `BrowserInstance` in a `ChartView` and would like to use/bind the same `BrowserInstance` across different view models (losely coupled), where I would like to write all of the chart controls/interactions logic.

I'm trying to use this `BrowserInstance` as a `DependencyProperty` but struggling to set,
`BrowserInstance.JavascriptObjectRepository.Settings.LegacyBindingEnabled = true;` resulting in `BrowserInstance` getting some `null` error?

There's a discussion thread [here](https://github.com/cefsharp/CefSharp/discussions/3609) but I'm not been able to follow :(
I am still learning and likely be looking in a wrong direction, can anyone please help me?

**Comment by amaitland** [comment, marked as answer]
>  It's unlikely you'd be able to set LegacyBindingEnabled in a ViewModel, it needs to be set before the underlying CEF browser is created and cannot be changed after.
> Subclassing ChromiumWebBrowser and setting the value in the constructor is likely the only option currently.

As I've discussed in #3609 you probably cannot set this in the `ViewModel`. You'll need to set it in the `View`, code behind is one option, or create your own class inheriting from `ChromiumWebBrowser` as I've suggested above.

**Reply by Dineshsing** [reply]
Thanks! Yes, having `LegacyBindingEnabled ` in the code behind is working - and I think there is no harm in having few bits of code in there. 
In fact - as you have mentioned, I'm now creating my own `CustomChromiumWebBrowser` inheriting from `ChromiumWebBrowser` and controling some parameters there. However, I can't really have a custom `CefSettings` as `Cef` is already being initialized to default settings from view.xaml I guess?! Perhaps, I may need to create another `CustomCefInitializer` helper class with my own settings like `cache`, `SchemeHandlerFactory` and calling this from `App.xaml.cs` - will see, but do let me know you thoughts!
Many thanks! :)

**Reply by amaitland** [reply]
> Perhaps, I may need to create another `CustomCefInitializer` helper class with my own settings like `cache`, `SchemeHandlerFactory` and calling this from `App.xaml.cs`

Calling `Cef.Initialzie` in `App.xaml.cs` is the place I'd recommend for `WPF`. There's an example at https://github.com/cefsharp/CefSharp.MinimalExample/blob/master/CefSharp.MinimalExample.Wpf/App.xaml.cs
